service/smithyprototype/lexruntimeservice/serde.go

```diff
@@ -0,0 +1,438 @@
+package lexruntimeservice
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws/protocol/json"
+	"github.com/aws/aws-sdk-go-v2/aws/protocol/rest"
+	"github.com/aws/aws-sdk-go-v2/service/smithyprototype/lexruntimeservice/types"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyHTTP "github.com/awslabs/smithy-go/transport/http"
+)
+
+type awsRestJson1_serializeOpDeleteSession struct{}
+
+// ID is the identifier for the middleware
+func (g *awsRestJson1_serializeOpDeleteSession) ID() string {
+	return "awsRestJson1_serializeOpDeleteSession"
+}
+
+// HandleSerialize will serialize the middleware input parameters to the provided input http request
+func (g *awsRestJson1_serializeOpDeleteSession) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyHTTP.Request)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown transport type %T", in.Request)
+	}
+
+	input, ok := in.Parameters.(*DeleteSessionInput)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown input parameters type %T", in.Parameters)
+	}
+
+	restEncoder := rest.NewEncoder(request.Request)
+
+	if err := awsRestJson1_serializeRestDeleteSessionInput(input, restEncoder); err != nil {
+		return out, metadata, err
+	}
+
+	if err := restEncoder.Encode(); err != nil {
+		return out, metadata, err
```

Comment:
Should these errors at the top level middleware be wrapped as `SerializationError`?